File: task4/Java/src/Task4.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class Task4 {

    public static void main(String[] args) {

        List<Integer> nums = new ArrayList<>();
        try {
        Scanner numbers = new Scanner(new File(args[0]));
        while (numbers.hasNextFloat()) {
            int x = numbers.nextInt();
            nums.add(x);
        }
        numbers.close();
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        }

        int average = nums.stream().reduce(0, Integer::sum) / nums.size();
        int moves = nums.stream().mapToInt(num -> Math.abs(num - average)).sum();

        System.out.println("Минимальное количество ходов: " + moves);
    }
}

File: task4/task4.cpp
#include <iostream>
#include <vector>
#include <fstream>

int main(int argc, char* argv[]) {

	if (argc != 2) {
		printf("Enter argument !\n");
		exit(1);
	}

	std::vector<int> nums;
	int num;
	std::ifstream numbers(argv[1]);
	if (numbers.is_open())
	{
		while (numbers >> num)
		{
			nums.push_back(num);
		}
	}
	numbers.close();

	int average = 0;
	for (int i = 0; i < nums.size(); i++) {
		average += nums[i];
	}
	average = average / nums.size();

	int moves = 0;
	for (int num : nums) {
		moves += std::abs(num - average);
	}

	std::cout << "Минимальное количество ходов: " << moves << std::endl;

	return 0;
}